### PR TITLE
DCMAW-6087 GitHub workflows

### PR DESCRIPTION
**Pull request by cgedrim-co**
### Add workflows and actions

- Workflows
  - on_push
  - on_pull_request
- Actions
  - bundle-reports
  - detect-os
  - gradle-check
  - gradle-connected-test
  - setup-runner

Resolves: DCMAW-6087

### CODE REVIEW

File: .github/workflows/on_pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: On Pull Request (develop)
+
+on:
+  pull_request:
+    branches:
+      - develop
```
Comment (Chuddington) [inline .github/workflows/on_pull_request.yml:6]: As part of following the gitflow branching strategy, we may want to include main in here as well

File: config/actions/detect-os/action.yml
Comment (Chuddington) [inline config/actions/detect-os/action.yml:1]: Do we want to provide outputs for this action, as opposed to appending the values to the environment?

https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#creating-an-action-metadata-file

File: .github/workflows/on_pull_request.yml (hunk 1 above)
Comment (cgedrim-co) [reply .github/workflows/on_pull_request.yml:6]: Yea, I was going to clarify that with Oli before I put it in

File: config/actions/setup-runner/action.yml
Comment (Chuddington) [inline config/actions/setup-runner/action.yml:1]: I'll just put it here as it seems to be quite common - we may want to split out linux / mac environment setup. As well as using provided inputs. Although, we wouldn't necessarily need inputs / environment variables if we have two separate actions that are then conditional based on the action's output

File: app/src/androidTest/java/uk/gov/onelogin/ExampleInstrumentedTest.kt [hunk 2]
```diff
@@ -6,19 +6,14 @@ import androidx.test.ext.junit.runners.AndroidJUnit4
...
- * See [testing documentation](http://d.android.com/tools/testing).
- */
 @RunWith(AndroidJUnit4::class)
 class ExampleInstrumentedTest {
     @Test
     fun useAppContext() {
         // Context of the app under test.
         val appContext = InstrumentationRegistry.getInstrumentation().targetContext
-        assertEquals("com.example.onelogin", appContext.packageName)
+        assertEquals("uk.gov.onelogin.staging", appContext.packageName)
```
[the remark this answers is not in the document]
Comment (cgedrim-co) [reply app/src/androidTest/java/uk/gov/onelogin/ExampleInstrumentedTest.kt:17]: Yup, but I've set the workflow to only run staging tests (as per instruction), and I didn't want to muddy the PR waters by adding flavor-specific tests in for now

File: config/actions/git-tag-bundles/action.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+name: 'Get the latest annotated tag'
+description: 'Gets the latest tag on a branch'
+
+inputs:
+  abb-paths:
+    description: 'Paths to build flavor abb'
```
[the remark this answers is not in the document]
Comment (cgedrim-co) [reply config/actions/git-tag-bundles/action.yml:6]: That's just my lysdexia... it should be `aab` 🤦

File: config/actions/git-tag-bundles/action.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  using: "composite"
+  steps:
+    - name: Git tag bundles
+      run: |
+        BUNDLES=($(echo $INPUT_ABB_PATHS | tr ";" "\n"))
+        
+        for BUNDLE in ${FLAVORS[@]}
+        do
+          IFS=- read -r FLAVOR ABB_PATH <<< $BUNDLE
+
```
[the remark this answers is not in the document]
Comment (cgedrim-co) [reply config/actions/git-tag-bundles/action.yml:30]: They're just strings that I manually pull apart in bash, so yes

**Review by Chuddington — APPROVED** [review]

File: config/styles/Vocab/Base/reject.txt
Comment (chrisdondig) [inline config/styles/Vocab/Base/reject.txt:1]: Is this empty file needed?

**Review by chrisdondig — APPROVED** [review]

File: config/styles/Vocab/Base/reject.txt
Comment (cgedrim-co) [reply config/styles/Vocab/Base/reject.txt:1]: Yes? I think that's one that @Chuddington added, but it's vocab for Vale

**Pull request merged**